### ISSUE TITLE
How to fix "DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it"

### ISSUE THREAD

**Issue by Hongbo-Miao** [hadolint/hadolint#724]
Originally asked at https://stackoverflow.com/questions/69684133/dl4006-warning-set-the-shell-option-o-pipefail-before-run-with-a-pipe-in-it

It would be great to have some guides for how to fix this, thanks! 😃

---

- [x] I searched existing issues before opening this one

### Steps to reproduce the behavior

**Output of `hadolint --version`

```bash
Haskell Dockerfile Linter 2.5.0-no-git
```

**Dockerfile**

```Dockerfile
FROM strimzi/kafka:0.20.1-kafka-2.6.0

USER root:root
RUN mkdir -p /opt/kafka/plugins/debezium
# Download, unpack, and place the debezium-connector-postgres folder into the /opt/kafka/plugins/debezium directory
RUN curl -s https://repo1.maven.org/maven2/io/debezium/debezium-connector-postgres/1.7.0.Final/debezium-connector-postgres-1.7.0.Final-plugin.tar.gz | tar xvz --transform 's/debezium-connector-postgres/debezium/' --directory /opt/kafka/plugins/
USER 1001
```

I got warning

> Dockerfile:6 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

I know I have a pipe `|` in the line started with `RUN`.

However, I still really don't know how to fix based on this warning.

**Comment by Hongbo-Miao** [comment]
Oh, just found in wiki at https://github.com/hadolint/hadolint/wiki/DL4006

Here is my working version:

```
FROM strimzi/kafka:0.20.1-kafka-2.6.0

USER root:root
RUN mkdir -p /opt/kafka/plugins/debezium
# Download, unpack, and place the debezium-connector-postgres folder into the /opt/kafka/plugins/debezium directory
SHELL ["/bin/bash", "-o", "pipefail", "-c"]
RUN curl -s https://repo1.maven.org/maven2/io/debezium/debezium-connector-postgres/1.7.0.Final/debezium-connector-postgres-1.7.0.Final-plugin.tar.gz | tar xvz --transform 's/debezium-connector-postgres/debezium/' --directory /opt/kafka/plugins/
USER 1001
```
